src/relua.cpp

```diff
@@ -0,0 +1,146 @@
+#include "relua.h"
+#include "mod.h"
+#include "openre.h"
+#include <cstdio>
+#include <filesystem>
+#include <lua.hpp>
+#include <memory>
+#include <string>
+
+namespace fs = std::filesystem;
+
+using namespace openre::modding;
+
+namespace openre::lua
+{
+    class LuaManager
+    {
+    private:
+        static inline std::unique_ptr<LuaManager> instance;
+        lua_State* _state;
+        std::vector<int> _subscriptionsTick;
+
+    public:
+        static LuaManager& get()
+        {
+            if (instance == nullptr)
+            {
+                instance = std::make_unique<LuaManager>();
+            }
+            return *instance;
+        }
+
+        LuaManager()
+        {
+            _state = luaL_newstate();
+            luaL_openlibs(_state);
+            setupGlobals();
+        }
+
+        LuaManager(const LuaManager&) = delete;
+
+        ~LuaManager()
+        {
+            lua_close(_state);
+        }
+
+        void runMod(Mod& mod)
+        {
+            mod.log("Starting");
+
+            const auto& scriptPath = mod.scriptPath;
+            auto result = luaL_dofile(_state, scriptPath.string().c_str());
+            if (result != LUA_OK)
+            {
+                auto errString = lua_tostring(_state, -1);
+                mod.log("Error: " + std::string(errString));
+            }
+        }
+
+        void callHooks(HookKind kind)
```

Comment:
I think we should make use of this parameter instead of invoking all ticks

Reply:
Not sure i understand

Reply:
`kind` parameter is not used.
At this point we only have undefined and ticks which is okay.

I assume the parameter acts like a filter. For example call all hooks of this kind.
But you're calling tick hooks ignoring the parameter.

Reply:
Yeah eventually I will add a `map<hook, vector<int>>` and it will use the parameter.